lib/zendesk_apps_support/validations/svg.rb

```diff
@@ -31,27 +39,43 @@ module Svg
       end
 
       class << self
+        def contains_embedded_bitmap?(markup)
+          Nokogiri::XML(markup).xpath('//image').any?
+        end
+
+        def rewrite_svg(svg, new_markup, package, errors)
+          warning_string = if contains_embedded_bitmap?(svg.read)
+                             'txt.apps.admin.warning.app_build.bitmap_in_svg'
+                           else
+                             'txt.apps.admin.warning.app_build.sanitised_svg'
+                           end
+
+          package.warnings << I18n.t(warning_string, svg: svg.relative_path)
+          IO.write(svg.absolute_path, new_markup)
+        rescue
+          errors << ValidationError.new(:dirty_svg, svg: svg.relative_path)
+        end
+
         def call(package)
           errors = []
 
           package.svg_files.each do |svg|
-            markup = Loofah.xml_fragment(svg.read)
-                           .scrub!(@strip_declaration)
-                           .scrub!(@strip_spaces_between_css_attrs)
-                           .to_xml.strip
-
-            clean_markup = Loofah.xml_fragment(markup)
-                                 .scrub!(:prune)
-                                 .scrub!(@empty_malformed_markup)
-                                 .to_xml
-
-            next if clean_markup == markup
-            begin
+            if contains_embedded_bitmap?(svg.read)
+              rewrite_svg(svg, PLACEHOLDER_SVG_MARKUP, package, errors)
```

Comment:
should we only clean the `<image>` part and still continue the next scrub steps?